File: 코드리뷰/한종호(lgwl)/3일차/코드리뷰_구간의합들.cpp
#include <iostream>
using namespace std;

int main()
{
	ios::sync_with_stdio(0);
	cin.tie(0); cout.tie(0);

	int N, M, sum, low = 0, high = 0, ans = 0;
	int arr[501];

	cin >> N >> M;
	for (int i = 0; i < N; i++)
		cin >> arr[i];

	sum = arr[0];
	while (low <= high && high < N)
	{
		if (sum == M)
			ans++;
		if (sum >= M)
			sum -= arr[low++];
		if (sum < M)
			sum += arr[++high];
	}

	cout << ans << "\n";

	return (0);
}

File: 코드리뷰/장문주(jmj2359)/3일차/코드리뷰_구간의합들.py
n, m = map(int,input().split())
arr = list(map(int,input().split()))
c = 0
for i in range(n-1):
    j=i
    sum = 0
    while (sum<=m and j<len(arr)):
        sum+=arr[j]
        j+=1
        if (sum==m):
            c+=1
print(c)
